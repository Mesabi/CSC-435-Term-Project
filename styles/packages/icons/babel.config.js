module.exports = {
  extends: '../../babel.config.js',
  ignore: ['**/*.d.ts'],
};
