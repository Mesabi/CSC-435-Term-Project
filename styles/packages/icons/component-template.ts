// @ts-check

import { IconData } from './processIcons';

export const getIconDataName = (name: string): string => {
  const lowercase = name.charAt(0).toLowerCase() + name.slice(1);
  return `${lowercase}Data`;
};

export const iconComponentTemplate = (
  name: string,
  images: IconData[],
): string => {
  const iconDataVariableName = getIconDataName(name);
  const componentName = name;
  const componentPropsName = `${componentName}Props`;
  return `import React from 'react';
import { IconProps } from '../iconType';
import { BaseIcon } from '../internal';
${images
  .map(image => `import ${image.id} from '${image.reactImportPath}';`)
  .join('\n')}

export const ${iconDataVariableName} = {${images
    .map(
      image => `
  '${image.variant}': {
    imageSrc: ${image.id} as string,
    width: ${image.width},
    height: ${image.height},
  },`,
    )
    .join('\n')}
};

export interface ${componentPropsName} extends IconProps {
  /**
   * Icon variant to use.
   * also provides default styling with the correct height and width
   **/
  variant?: ${images.map(image => `'${image.variant}'`).join(' | ')};
}

export const ${componentName}: React.FC<${componentPropsName}> = ({
  variant = '${images[0].variant}',
  ...rest
}) => {
  const image = ${iconDataVariableName}[variant];

  return (
    <BaseIcon
      width={image.width}
      height={image.height}
      src={image.imageSrc}
      {...rest}
    />
  );
};
`;
};
