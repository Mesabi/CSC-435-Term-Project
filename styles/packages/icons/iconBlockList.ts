import { IconData } from './processIcons';

/** This list includes all icons seen as not including any visual output */
const iconBlockedIds = [
  'Access218_16x16_4',
  'Access219_16x16_4',
  'Access220_16x16_4',
  'Access221_16x16_4',
  'Access222_16x16_4',
  'Access223_16x16_4',
  'Access224_16x16_4',
  'Access225_16x16_4',
  'Access226_16x16_4',
  'Access227_16x16_4',
  'Access228_16x16_4',
  'Access229_16x16_4',
  'Access230_16x16_4',
  'Appwiz1501_16x16_4',
  'Appwiz1502_16x16_4',
  'Awfext326050_32x32_4',
  'Awfxcg321302_16x16_4',
  'Awfxcg321303_16x16_4',
  'Awfxcg321305_16x16_4',
  'Dial16x16_4',
  'Explorer104_32x32_4',
  'Filefind3_16x16_4',
  'Filepick_16x16_4',
  'Filexfer129_16x16_4',
  'Filexfer130_16x16_4',
  'Gcdef108_32x32_4',
  'Gcdef109_32x32_4',
  'Gcdef110_32x32_4',
  'Gcdef111_32x32_4',
  'Gcdef112_32x32_4',
  'Gcdef113_32x32_4',
  'Gcdef114_32x32_4',
  'Gcdef115_32x32_4',
  'Gcdef116_32x32_4',
  'Gcdef117_32x32_4',
  'Gcdef122_32x32_4',
  'Gcdef124_32x32_4',
  'Lights101_32x32_4',
  'Lights102_32x32_4',
  'Lights103_32x32_4',
  'Mshtml32539_16x16_4',
  'Mshtml32544_32x32_4',
  'Mshtml32545_32x32_4',
  'Shdocvw261_32x32_4',
  'Shdocvw274_32x32_4',
  'Shdocvw275_32x32_4',
];

/** Returns true if icon is not marked as blocked */
export const filterBlockedIcons = (icon: IconData): boolean =>
  !iconBlockedIds.includes(icon.id);
