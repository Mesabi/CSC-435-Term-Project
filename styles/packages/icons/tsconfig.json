{
  "extends": "../../tsconfig.json",
  "compilerOptions": {
    "allowJs": true,
    "outDir": "dist/@types",
    "jsx": "react",
    "incremental": true,
    "declaration": true,
    "emitDeclarationOnly": true
  },
  "include": ["src"]
}
