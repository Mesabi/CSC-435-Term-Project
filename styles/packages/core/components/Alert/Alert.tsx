import React, { useEffect, forwardRef } from 'react';
import styled from '@xstyled/styled-components';
import { User3, User2, User4, User5 } from '@react95/icons';

import Modal, { ModalProps } from '../Modal/Modal';

import sound from './assets/chord.mp3';

export type AlertType = 'error' | 'info' | 'question' | 'warning';

const RenderImage: React.FC<{ option: string }> = ({ option }) => {
  switch (option) {
    case 'info':
      return <User5 variant="32x32_4" />;
    case 'question':
      return <User3 variant="32x32_4" />;
    case 'warning':
      return <User2 variant="32x32_4" />;
    case 'error':
    default:
      return <User4 variant="32x32_4" />;
  }
};

const Message = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
`;

const IconWrapper = styled.div`
  padding: 7 15 7 7;
`;

const Dialog = styled.div`
  display: flex;
  flex-direction: row;
`;

export type AlertProps = Omit<ModalProps, 'closeModal'> & {
  message: string;
  closeAlert: ModalProps['closeModal'];
  hasSound?: boolean;
  type?: AlertType;
};

const Alert = forwardRef<HTMLDivElement, AlertProps>(
  ({ type = 'error', message, closeAlert, hasSound, ...rest }, ref) => {
    if (hasSound) {
      useEffect(() => {
        const audio = new Audio(sound);
        audio.play();
      }, []);
    }

    return (
      <Modal
        closeModal={closeAlert}
        height="120"
        hasWindowButton={false}
        {...rest}
        ref={ref}
      >
        <Dialog>
          <IconWrapper>
            <RenderImage option={type} />
          </IconWrapper>
          <Message>{message}</Message>
        </Dialog>
      </Modal>
    );
  },
);

Alert.displayName = 'Alert';

Alert.defaultProps = {
  type: 'error',
  buttons: [{ value: 'OK', onClick: () => {} }],
  closeAlert: () => {},
  buttonsAlignment: 'center',
  defaultPosition: {
    x:
      typeof window == 'undefined'
        ? 0
        : Math.floor(window.innerWidth / 2) - 150,
    y:
      typeof window == 'undefined'
        ? 0
        : Math.floor(window.innerHeight / 2) - 80,
  },
  hasSound: false,
};

export default Alert;
