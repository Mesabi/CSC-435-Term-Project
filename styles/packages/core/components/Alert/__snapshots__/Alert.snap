// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Alert /> Snapshots should match snapshot with error type 1`] = `
.c6 {
  background-color: #c3c7cb;
  padding: 7px 20px 5px;
  border: none;
  color: #000000;
  font-size: 12px;
  box-shadow: inset 1px 1px 0px 1px #ffffff, inset 0 0 0 1px #868a8e, 1px 1px 0 0px #000000;
}

.c6:disabled {
  color: #868a8e;
}

.c6:focus {
  outline: 1px dotted #000000;
  outline-offset: -5px;
  box-shadow: inset 1px 1px 0px 1px #ffffff, inset -0.5px -0.5px 0px 1px #868a8e, 1px 1px 0 1px #000000;
}

.c6:active {
  padding: 8px 20px 4px;
  outline: 1px dotted #000000;
  outline-offset: -5px;
  box-shadow: inset 0 0 0 1px #868a8e, 0 0 0 1px #000000;
}

.c1 {
  color: #fefefe;
  width: auto;
  height: auto;
  background-color: #c3c7cb;
  --x-shadow: inset 0.5px 0.5px 0px 0.5px #ffffff, inset 0 0 0 1px #868a8e, 1px 0px 0 0px #000000, 0px 1px 0 0px #000000, 1px 1px 0 0px #000000;
  box-shadow: var(--x-ring-shadow,0 0 #0000),var(--x-shadow);
}

.c4 {
  list-style: none;
  padding: 0;
  margin: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-left: 2px;
  padding: 0;
  width: 17px;
  height: 14px;
  min-width: 0;
  font-weight: 600;
  font-size: 10px;
}

.c7:active {
  padding: 1px 0 0 1px;
  outline: none;
}

.c7:first-child {
  margin-right: 0;
}

.c7:focus {
  box-shadow: inset 1px 1px 0px 1px #ffffff, inset -1px -1px 0px 1px #868a8e;
}

.c2 {
  height: 18px;
  margin-bottom: 2px;
  padding: 2px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-shadow: none;
  background: #000e7a;
}

.c2 img {
  width: 15px;
  height: 14px;
  margin-right: 4px;
}

.c3 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  font-weight: bold;
  line-height: 1.4em;
  margin: 0;
  font-size: 1em;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: fixed;
  padding: 2px 2px 8px;
  top: 50px;
  background-color: #c3c7cb;
  box-shadow: inset 1px 1px 0px 1px #ffffff, inset 0 0 0 1px #868a8e, 1px 1px 0 1px #000000;
  width: auto;
  height: 120px;
  z-index: 1;
}

.c8 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 6px;
}

.c12 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0 6px 6px 6px;
}

.c12 .c5 {
  margin-right: 6px;
  min-width: 70px;
}

.c12 .c5:last-child {
  margin-right: 0;
}

.c11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c10 {
  padding: 7px 15px 7px 7px;
}

.c9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

<div>
  <div
    class="c0 react-draggable"
    height="120"
    style="transform: translate(362px,304px);"
  >
    <div
      class="c1 c2"
      color="headerText"
    >
      <div
        class="c3 draggable"
      >
        Error
      </div>
      <ul
        class="c4"
      >
        <li
          class="c5 c6 c7"
        >
          ?
        </li>
        <li
          class="c5 c6 c7"
        >
          X
        </li>
      </ul>
    </div>
    <div
      class="c8"
    >
      <div
        class="c9"
      >
        <div
          class="c10"
        >
          <img
            height="32"
            src="test-file-stub"
            width="32"
          />
        </div>
        <div
          class="c11"
        >
          Error
        </div>
      </div>
    </div>
    <div
      class="c12"
    >
      <button
        class="c5 c6"
        value="OK"
      >
        OK
      </button>
    </div>
  </div>
</div>
`;

exports[`<Alert /> Snapshots should match snapshot with info type 1`] = `
.c6 {
  background-color: #c3c7cb;
  padding: 7px 20px 5px;
  border: none;
  color: #000000;
  font-size: 12px;
  box-shadow: inset 1px 1px 0px 1px #ffffff, inset 0 0 0 1px #868a8e, 1px 1px 0 0px #000000;
}

.c6:disabled {
  color: #868a8e;
}

.c6:focus {
  outline: 1px dotted #000000;
  outline-offset: -5px;
  box-shadow: inset 1px 1px 0px 1px #ffffff, inset -0.5px -0.5px 0px 1px #868a8e, 1px 1px 0 1px #000000;
}

.c6:active {
  padding: 8px 20px 4px;
  outline: 1px dotted #000000;
  outline-offset: -5px;
  box-shadow: inset 0 0 0 1px #868a8e, 0 0 0 1px #000000;
}

.c1 {
  color: #fefefe;
  width: auto;
  height: auto;
  background-color: #c3c7cb;
  --x-shadow: inset 0.5px 0.5px 0px 0.5px #ffffff, inset 0 0 0 1px #868a8e, 1px 0px 0 0px #000000, 0px 1px 0 0px #000000, 1px 1px 0 0px #000000;
  box-shadow: var(--x-ring-shadow,0 0 #0000),var(--x-shadow);
}

.c4 {
  list-style: none;
  padding: 0;
  margin: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-left: 2px;
  padding: 0;
  width: 17px;
  height: 14px;
  min-width: 0;
  font-weight: 600;
  font-size: 10px;
}

.c7:active {
  padding: 1px 0 0 1px;
  outline: none;
}

.c7:first-child {
  margin-right: 0;
}

.c7:focus {
  box-shadow: inset 1px 1px 0px 1px #ffffff, inset -1px -1px 0px 1px #868a8e;
}

.c2 {
  height: 18px;
  margin-bottom: 2px;
  padding: 2px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-shadow: none;
  background: #000e7a;
}

.c2 img {
  width: 15px;
  height: 14px;
  margin-right: 4px;
}

.c3 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  font-weight: bold;
  line-height: 1.4em;
  margin: 0;
  font-size: 1em;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: fixed;
  padding: 2px 2px 8px;
  top: 50px;
  background-color: #c3c7cb;
  box-shadow: inset 1px 1px 0px 1px #ffffff, inset 0 0 0 1px #868a8e, 1px 1px 0 1px #000000;
  width: auto;
  height: 120px;
  z-index: 1;
}

.c8 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 6px;
}

.c12 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0 6px 6px 6px;
}

.c12 .c5 {
  margin-right: 6px;
  min-width: 70px;
}

.c12 .c5:last-child {
  margin-right: 0;
}

.c11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c10 {
  padding: 7px 15px 7px 7px;
}

.c9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

<div>
  <div
    class="c0 react-draggable"
    height="120"
    style="transform: translate(362px,304px);"
  >
    <div
      class="c1 c2"
      color="headerText"
    >
      <div
        class="c3 draggable"
      >
        Info
      </div>
      <ul
        class="c4"
      >
        <li
          class="c5 c6 c7"
        >
          ?
        </li>
        <li
          class="c5 c6 c7"
        >
          X
        </li>
      </ul>
    </div>
    <div
      class="c8"
    >
      <div
        class="c9"
      >
        <div
          class="c10"
        >
          <img
            height="32"
            src="test-file-stub"
            width="32"
          />
        </div>
        <div
          class="c11"
        >
          Info
        </div>
      </div>
    </div>
    <div
      class="c12"
    >
      <button
        class="c5 c6"
        value="OK"
      >
        OK
      </button>
    </div>
  </div>
</div>
`;

exports[`<Alert /> Snapshots should match snapshot with question type 1`] = `
.c6 {
  background-color: #c3c7cb;
  padding: 7px 20px 5px;
  border: none;
  color: #000000;
  font-size: 12px;
  box-shadow: inset 1px 1px 0px 1px #ffffff, inset 0 0 0 1px #868a8e, 1px 1px 0 0px #000000;
}

.c6:disabled {
  color: #868a8e;
}

.c6:focus {
  outline: 1px dotted #000000;
  outline-offset: -5px;
  box-shadow: inset 1px 1px 0px 1px #ffffff, inset -0.5px -0.5px 0px 1px #868a8e, 1px 1px 0 1px #000000;
}

.c6:active {
  padding: 8px 20px 4px;
  outline: 1px dotted #000000;
  outline-offset: -5px;
  box-shadow: inset 0 0 0 1px #868a8e, 0 0 0 1px #000000;
}

.c1 {
  color: #fefefe;
  width: auto;
  height: auto;
  background-color: #c3c7cb;
  --x-shadow: inset 0.5px 0.5px 0px 0.5px #ffffff, inset 0 0 0 1px #868a8e, 1px 0px 0 0px #000000, 0px 1px 0 0px #000000, 1px 1px 0 0px #000000;
  box-shadow: var(--x-ring-shadow,0 0 #0000),var(--x-shadow);
}

.c4 {
  list-style: none;
  padding: 0;
  margin: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-left: 2px;
  padding: 0;
  width: 17px;
  height: 14px;
  min-width: 0;
  font-weight: 600;
  font-size: 10px;
}

.c7:active {
  padding: 1px 0 0 1px;
  outline: none;
}

.c7:first-child {
  margin-right: 0;
}

.c7:focus {
  box-shadow: inset 1px 1px 0px 1px #ffffff, inset -1px -1px 0px 1px #868a8e;
}

.c2 {
  height: 18px;
  margin-bottom: 2px;
  padding: 2px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-shadow: none;
  background: #000e7a;
}

.c2 img {
  width: 15px;
  height: 14px;
  margin-right: 4px;
}

.c3 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  font-weight: bold;
  line-height: 1.4em;
  margin: 0;
  font-size: 1em;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: fixed;
  padding: 2px 2px 8px;
  top: 50px;
  background-color: #c3c7cb;
  box-shadow: inset 1px 1px 0px 1px #ffffff, inset 0 0 0 1px #868a8e, 1px 1px 0 1px #000000;
  width: auto;
  height: 120px;
  z-index: 1;
}

.c8 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 6px;
}

.c12 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0 6px 6px 6px;
}

.c12 .c5 {
  margin-right: 6px;
  min-width: 70px;
}

.c12 .c5:last-child {
  margin-right: 0;
}

.c11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c10 {
  padding: 7px 15px 7px 7px;
}

.c9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

<div>
  <div
    class="c0 react-draggable"
    height="120"
    style="transform: translate(362px,304px);"
  >
    <div
      class="c1 c2"
      color="headerText"
    >
      <div
        class="c3 draggable"
      >
        question
      </div>
      <ul
        class="c4"
      >
        <li
          class="c5 c6 c7"
        >
          ?
        </li>
        <li
          class="c5 c6 c7"
        >
          X
        </li>
      </ul>
    </div>
    <div
      class="c8"
    >
      <div
        class="c9"
      >
        <div
          class="c10"
        >
          <img
            height="32"
            src="test-file-stub"
            width="32"
          />
        </div>
        <div
          class="c11"
        >
          question
        </div>
      </div>
    </div>
    <div
      class="c12"
    >
      <button
        class="c5 c6"
        value="OK"
      >
        OK
      </button>
    </div>
  </div>
</div>
`;

exports[`<Alert /> Snapshots should match snapshot with warning type 1`] = `
.c6 {
  background-color: #c3c7cb;
  padding: 7px 20px 5px;
  border: none;
  color: #000000;
  font-size: 12px;
  box-shadow: inset 1px 1px 0px 1px #ffffff, inset 0 0 0 1px #868a8e, 1px 1px 0 0px #000000;
}

.c6:disabled {
  color: #868a8e;
}

.c6:focus {
  outline: 1px dotted #000000;
  outline-offset: -5px;
  box-shadow: inset 1px 1px 0px 1px #ffffff, inset -0.5px -0.5px 0px 1px #868a8e, 1px 1px 0 1px #000000;
}

.c6:active {
  padding: 8px 20px 4px;
  outline: 1px dotted #000000;
  outline-offset: -5px;
  box-shadow: inset 0 0 0 1px #868a8e, 0 0 0 1px #000000;
}

.c1 {
  color: #fefefe;
  width: auto;
  height: auto;
  background-color: #c3c7cb;
  --x-shadow: inset 0.5px 0.5px 0px 0.5px #ffffff, inset 0 0 0 1px #868a8e, 1px 0px 0 0px #000000, 0px 1px 0 0px #000000, 1px 1px 0 0px #000000;
  box-shadow: var(--x-ring-shadow,0 0 #0000),var(--x-shadow);
}

.c4 {
  list-style: none;
  padding: 0;
  margin: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-left: 2px;
  padding: 0;
  width: 17px;
  height: 14px;
  min-width: 0;
  font-weight: 600;
  font-size: 10px;
}

.c7:active {
  padding: 1px 0 0 1px;
  outline: none;
}

.c7:first-child {
  margin-right: 0;
}

.c7:focus {
  box-shadow: inset 1px 1px 0px 1px #ffffff, inset -1px -1px 0px 1px #868a8e;
}

.c2 {
  height: 18px;
  margin-bottom: 2px;
  padding: 2px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-shadow: none;
  background: #000e7a;
}

.c2 img {
  width: 15px;
  height: 14px;
  margin-right: 4px;
}

.c3 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  font-weight: bold;
  line-height: 1.4em;
  margin: 0;
  font-size: 1em;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: fixed;
  padding: 2px 2px 8px;
  top: 50px;
  background-color: #c3c7cb;
  box-shadow: inset 1px 1px 0px 1px #ffffff, inset 0 0 0 1px #868a8e, 1px 1px 0 1px #000000;
  width: auto;
  height: 120px;
  z-index: 1;
}

.c8 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 6px;
}

.c12 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0 6px 6px 6px;
}

.c12 .c5 {
  margin-right: 6px;
  min-width: 70px;
}

.c12 .c5:last-child {
  margin-right: 0;
}

.c11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c10 {
  padding: 7px 15px 7px 7px;
}

.c9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

<div>
  <div
    class="c0 react-draggable"
    height="120"
    style="transform: translate(362px,304px);"
  >
    <div
      class="c1 c2"
      color="headerText"
    >
      <div
        class="c3 draggable"
      >
        Warning
      </div>
      <ul
        class="c4"
      >
        <li
          class="c5 c6 c7"
        >
          ?
        </li>
        <li
          class="c5 c6 c7"
        >
          X
        </li>
      </ul>
    </div>
    <div
      class="c8"
    >
      <div
        class="c9"
      >
        <div
          class="c10"
        >
          <img
            height="32"
            src="test-file-stub"
            width="32"
          />
        </div>
        <div
          class="c11"
        >
          Warning
        </div>
      </div>
    </div>
    <div
      class="c12"
    >
      <button
        class="c5 c6"
        value="OK"
      >
        OK
      </button>
    </div>
  </div>
</div>
`;
